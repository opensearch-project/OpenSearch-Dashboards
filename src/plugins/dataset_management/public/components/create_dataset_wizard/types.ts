/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface MatchedItem {
  name: string;
  tags: Tag[];
  item: {
    name: string;
    backing_indices?: string[];
    timestamp_field?: string;
    indices?: string[];
    aliases?: string[];
    attributes?: ResolveIndexResponseItemIndexAttrs[];
    data_stream?: string;
  };
}

export interface ResolveIndexResponse {
  indices?: ResolveIndexResponseItemIndex[];
  aliases?: ResolveIndexResponseItemAlias[];
  data_streams?: ResolveIndexResponseItemDataStream[];
}

export interface ResolveIndexResponseItem {
  name: string;
}

export interface ResolveIndexResponseItemDataStream extends ResolveIndexResponseItem {
  backing_indices: string[];
  timestamp_field: string;
}

export interface ResolveIndexResponseItemAlias extends ResolveIndexResponseItem {
  indices: string[];
}

export interface ResolveIndexResponseItemIndex extends ResolveIndexResponseItem {
  aliases?: string[];
  attributes?: ResolveIndexResponseItemIndexAttrs[];
  data_stream?: string;
}

export enum ResolveIndexResponseItemIndexAttrs {
  OPEN = 'open',
  CLOSED = 'closed',
  HIDDEN = 'hidden',
  FROZEN = 'frozen',
}

export interface Tag {
  name: string;
  key: string;
  color: string;
}

export interface StepInfo {
  totalStepNumber: number;
  currentStepNumber: number;
}

export interface DataSourceTableItem {
  id: string;
  type: string;
  title: string;
  sort: string;
  checked?: 'on' | 'off';
  label: string;
  datasourceversion: string;
  installedplugins: string[];
  engine: string;
  relatedDataSourceConnection?: DataSourceTableItem[];
  append?: any;
  prepend?: any;
  parentId?: string;
  disabled?: boolean;
}
