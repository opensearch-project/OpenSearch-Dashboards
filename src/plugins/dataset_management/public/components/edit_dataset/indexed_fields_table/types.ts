/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { IFieldType } from '../../../../../../plugins/data/public';

export interface IndexedFieldItem extends IFieldType {
  info: string[];
  excluded: boolean;
}
