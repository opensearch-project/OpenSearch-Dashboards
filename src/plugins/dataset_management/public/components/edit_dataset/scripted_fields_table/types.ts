/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/** @internal **/
export interface ScriptedFieldItem {
  name: string;
  lang: string;
  script: string;
}
