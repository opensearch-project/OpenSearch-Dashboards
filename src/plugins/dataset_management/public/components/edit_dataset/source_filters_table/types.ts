/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/** @internal **/
export interface SourceFiltersTableFilter {
  value: string;
  clientId: string | number;
}
