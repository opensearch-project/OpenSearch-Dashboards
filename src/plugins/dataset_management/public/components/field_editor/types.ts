/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { ReactText } from 'react';
import { Query } from 'src/plugins/data/public';
import { HttpStart } from 'src/core/public';

export interface Sample {
  input: ReactText | ReactText[];
  output: string;
}

export interface ExecuteScriptParams {
  name: string;
  script: string;
  indexPatternTitle: string;
  query?: Query['query'];
  additionalFields?: string[];
  http: HttpStart;
  dataSourceId?: string;
}

export interface ExecuteScriptResult {
  status: number;
  hits?: { hits: any[] };
  error?: any;
}

export type ExecuteScript = (params: ExecuteScriptParams) => Promise<ExecuteScriptResult>;
