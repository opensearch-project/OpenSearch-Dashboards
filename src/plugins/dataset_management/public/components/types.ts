/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface DatasetCreationOption {
  text: string;
  description?: string;
  onClick: () => void;
}

export interface DatasetTableItem {
  id: string;
  title: string;
  default: boolean;
  tag?: string[];
  sort: string;
}
