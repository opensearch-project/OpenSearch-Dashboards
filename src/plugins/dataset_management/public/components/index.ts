/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { DatasetTableWithRouter } from './dataset_table';
export {
  EditDataset,
  EditDatasetContainer,
  CreateEditField,
  CreateEditFieldContainer,
} from './edit_dataset';
export { CreateDatasetWizardWithRouter } from './create_dataset_wizard';
