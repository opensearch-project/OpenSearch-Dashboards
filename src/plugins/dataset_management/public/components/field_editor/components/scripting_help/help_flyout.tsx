/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from 'react';
import { EuiFlyout, EuiFlyoutBody, EuiTabbedContent } from '@elastic/eui';

import { ScriptingSyntax } from './scripting_syntax';
import { TestScript } from './test_script';

import { ExecuteScript } from '../../types';
import { DataView } from '../../../../../../data/public';

interface ScriptingHelpFlyoutProps {
  dataset: DataView;
  lang: string;
  name?: string;
  script?: string;
  executeScript: ExecuteScript;
  isVisible: boolean;
  onClose: () => void;
}

export const ScriptingHelpFlyout: React.FC<ScriptingHelpFlyoutProps> = ({
  isVisible = false,
  onClose = () => {},
  dataset,
  lang,
  name,
  script,
  executeScript,
}) => {
  const tabs = [
    {
      id: 'syntax',
      name: 'Syntax',
      ['data-test-subj']: 'syntaxTab',
      content: <ScriptingSyntax />,
    },
    {
      id: 'test',
      name: 'Preview results',
      ['data-test-subj']: 'testTab',
      content: (
        <TestScript
          dataset={dataset}
          lang={lang}
          name={name}
          script={script}
          executeScript={executeScript}
        />
      ),
    },
  ];

  return isVisible ? (
    <EuiFlyout onClose={onClose} data-test-subj="scriptedFieldsHelpFlyout">
      <EuiFlyoutBody>
        <EuiTabbedContent tabs={tabs} initialSelectedTab={tabs[0]} size="s" />
      </EuiFlyoutBody>
    </EuiFlyout>
  ) : null;
};

ScriptingHelpFlyout.displayName = 'ScriptingHelpFlyout';
