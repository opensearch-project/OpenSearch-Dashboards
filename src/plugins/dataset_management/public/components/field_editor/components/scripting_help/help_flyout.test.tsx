/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from 'react';
import { shallow } from 'enzyme';

import { ScriptingHelpFlyout } from './help_flyout';

import { DataView } from '../../../../../../data/public';

import { ExecuteScript } from '../../types';

jest.mock('./test_script', () => ({
  TestScript: () => {
    return `<div>mockTestScript</div>`;
  },
}));

const indexPatternMock = {} as DataView;

describe('ScriptingHelpFlyout', () => {
  it('should render normally', async () => {
    const component = shallow(
      <ScriptingHelpFlyout
        isVisible={true}
        dataset={indexPatternMock}
        lang="painless"
        executeScript={((() => {}) as unknown) as ExecuteScript}
        onClose={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render nothing if not visible', async () => {
    const component = shallow(
      <ScriptingHelpFlyout
        isVisible={true}
        dataset={indexPatternMock}
        lang="painless"
        executeScript={((() => {}) as unknown) as ExecuteScript}
        onClose={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
