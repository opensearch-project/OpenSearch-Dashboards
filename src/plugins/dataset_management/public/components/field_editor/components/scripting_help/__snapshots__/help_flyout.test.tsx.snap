// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptingHelpFlyout should render normally 1`] = `
<EuiFlyout
  data-test-subj="scriptedFieldsHelpFlyout"
  onClose={[Function]}
>
  <EuiFlyoutBody>
    <EuiTabbedContent
      autoFocus="initial"
      initialSelectedTab={
        Object {
          "content": <ScriptingSyntax />,
          "data-test-subj": "syntaxTab",
          "id": "syntax",
          "name": "Syntax",
        }
      }
      preserveTabContent={false}
      size="s"
      tabs={
        Array [
          Object {
            "content": <ScriptingSyntax />,
            "data-test-subj": "syntaxTab",
            "id": "syntax",
            "name": "Syntax",
          },
          Object {
            "content": <TestScript
              dataset={Object {}}
              executeScript={[Function]}
              lang="painless"
            />,
            "data-test-subj": "testTab",
            "id": "test",
            "name": "Preview results",
          },
        ]
      }
    />
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`ScriptingHelpFlyout should render nothing if not visible 1`] = `
<EuiFlyout
  data-test-subj="scriptedFieldsHelpFlyout"
  onClose={[Function]}
>
  <EuiFlyoutBody>
    <EuiTabbedContent
      autoFocus="initial"
      initialSelectedTab={
        Object {
          "content": <ScriptingSyntax />,
          "data-test-subj": "syntaxTab",
          "id": "syntax",
          "name": "Syntax",
        }
      }
      preserveTabContent={false}
      size="s"
      tabs={
        Array [
          Object {
            "content": <ScriptingSyntax />,
            "data-test-subj": "syntaxTab",
            "id": "syntax",
            "name": "Syntax",
          },
          Object {
            "content": <TestScript
              dataset={Object {}}
              executeScript={[Function]}
              lang="painless"
            />,
            "data-test-subj": "testTab",
            "id": "test",
            "name": "Preview results",
          },
        ]
      }
    />
  </EuiFlyoutBody>
</EuiFlyout>
`;
